aggregate_debug_reporting.md

```diff
@@ -0,0 +1,241 @@
+**Table of Contents**
+
+- [Aggregate Debug Reporting](#aggregate-debug-reporting)
+  - [Introduction](#introduction)
+  - [API changes](#api-changes)
+    - [Opting-in aggregate debug reporting](#opting-in-aggregate-debug-reporting)
+    - [Aggregatable debug reports](#aggregatable-debug-reports)
+      - [Encrypted payload](#encrypted-payload)
+  - [Privacy and Security](#privacy-and-security)
+    - [Contribution bounding and budgeting](#contribution-bounding-and-budgeting)
+    - [Make count of reports deterministic with null reports](#make-count-of-reports-deterministic-with-null-reports)
+    - [Rate limits](#rate-limits)
+    - [No reporting delay](#no-reporting-delay)
+  - [Future considerations](#future-considerations)
+    - [Report verification with debug context ID](#report-verification-with-debug-context-id)
+    - [Application to aggregate error reporting for the Private Aggregation API](#application-to-aggregate-error-reporting-for-the-private-aggregation-api)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Introduction
+
+The [transitional debugging reports](https://github.com/WICG/attribution-reporting-api/blob/main/EVENT.md#optional-transitional-debugging-reports),
+which are currently supported by the Attribution Reporting API to facilitate
+testing, will be deprecated alongside third-party cookie deprecation.
+
+Here we propose the aggregate debug reporting framework to allow developers to
+measure the performance of the Attribution Reporting API post third-party
+cookie deprecation. The aggregate debug reporting utilizes the mechanism
+introduced in the [Aggregate Attribution Reporting](https://github.com/WICG/attribution-reporting-api/blob/main/AGGREGATE.md)
+to report debug data in aggregate. To prevent leakage, the cross-site data in
+the aggregatable reports would be encrypted to ensure it can only be processed
+by the [aggregation service](https://github.com/WICG/attribution-reporting-api/blob/main/AGGREGATION_SERVICE_TEE.md).
+During processing, this service will add noise and impose limits on how many
+queries can be performed.
+
+## API changes
+
+### Opting-in aggregate debug reporting
+
+The [source registration](https://github.com/WICG/attribution-reporting-api/blob/main/EVENT.md#registering-attribution-sources)
+and [trigger registration](https://github.com/WICG/attribution-reporting-api/blob/main/EVENT.md#triggering-attribution)
+will accept an optional dictionary field:
+
+```jsonc
+{
+  ..., // existing fields
+
+  "aggregatable_debug_reporting": {
+     "budget": 1024, // required, source registration only
+     "key_piece": "0x120", // required, source- or trigger-side key piece
+     "debug_data": [    
+        {  
+          "types": ["source-destination-limit",  "source-destination-rate-limit"], // required to be non-empty
+          "key_piece": "0x1", // required
+          "value": 123 // required
+       },
+       {  
+         "types": ["unspecified"],
```

Comment:
I find the term `unspecified` to be a bit vague. I think it could be misinterpreted as "do anything". I wonder if a more conventional wildcard makes more sense, e.g. `*`, which is also a bit more concise and will pretty clearly never conflict with an actual debug report type.